byzcoin/service.go

```diff
@@ -704,6 +705,13 @@ func (s *Service) GetReadOnlyStateTrie(scID skipchain.SkipBlockID) (ReadOnlyStat
 	return s.getStateTrie(scID)
 }
 
+func (s *Service) hasStateTrie(id skipchain.SkipBlockID) bool {
```

Comment:
On line 719 below, it seems like s.storage.Mutex guard access to this map. This is not well documented, and messy. Let's fix it.

So: s/omniStorage/bcStorage/, use less generic name, s.storage.Mutex.Lock -> s.storage.Lock (same for unlock). Make a new mutex for stateTrie, put it NEXT TO stateTrie in the type defn.